pkg/generator/password_reconciler.go

```diff
@@ -115,37 +123,96 @@ func (r *PasswordReconciler) createSecret(ctx context.Context, password *sgv1alp
 	return reconcile.Result{}, nil
 }
 
-type passwordParams struct {
-	Length int `yaml:"length"`
+func (r *PasswordReconciler) generate(password *sgv1alpha1.Password) (string, error) {
+	passwordVal := localGeneratePassword(&password.Spec)
+	return passwordVal, nil
 }
 
-func (r *PasswordReconciler) generate(password *sgv1alpha1.Password) (string, error) {
-	gen := cfgtypes.NewPasswordGenerator()
+func localGeneratePassword(spec *sgv1alpha1.PasswordSpec) string {
+
+	var password strings.Builder
+
+	//Set symbol character
+	for i := 0; i < spec.Symbols; i++ {
+		value, _ := randomElement(spec.SymbolCharSet)
+		password.WriteString(value)
+	}
+
+	//Set digit
+	for i := 0; i < spec.Digits; i++ {
+		value, _ := randomElement(digitSet)
+		password.WriteString(value)
+	}
+
+	//Set uppercase
+	for i := 0; i < spec.UppercaseLetters; i++ {
+		value, _ := randomElement(upperCharSet)
+		password.WriteString(value)
+	}
 
-	genParams := passwordParams{Length: password.Spec.Length}
-	if genParams.Length == 0 {
-		genParams.Length = 40
+	//Set lowercase
+	for i := 0; i < spec.LowercaseLetters; i++ {
+		value, _ := randomElement(lowerCharSet)
+		password.WriteString(value)
 	}
 
-	passwordVal, err := gen.Generate(genParams)
+	var allCharSet = lowerCharSet + upperCharSet + digitSet
```

Comment:
Is this missing the symbolcharset? 

```suggestion
	var allCharSet = lowerCharSet + upperCharSet + digitSet + spec.SymbolCharSet
```

Reply:
I didn't add spec.SymbolCharSet because I don't want to add extra SymbolCharset to fill in the generated password. Systems don't allow too many symbols. So if the user specifies x symbols, the user gets x symbols. not more.

Reply:
ok cool, seems reasonable to me. Just wanted to double check the reasoning 👍